app/models/user.rb

```diff
@@ -210,6 +210,23 @@ def self.non_student_known_roles
     known_roles - ["student"]
   end
 
+  def guessed_preferred_confirmed_email
+    # A heuristic for guessing the user's preferred confirmed email.  Assumes that
+    # emails that were manually entered are more preferred than those that were
+    # added via a social login. Manually-entered emails trigger confirmation emails,
+    # so those emails have the confirmation sent at timestamp.
+
+    (
+      email_addresses.verified
+                     .where{confirmation_sent_at != nil} # manually verified
+                     .order{created_at.asc}
+                     .first ||
+      email_addresses.verified
+                     .order{created_at.asc}
+                     .first
+    ).try(:value)
+  end
+
   protected
```

Comment:
I'm guessing that there's no real way to detect them, but it seems to me that the last manually entered address would be preferred.  I can see users maybe adding a new email address when they get one, and not removing the previous one.